server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -267,10 +267,10 @@ private void cleanUpDeletedTableDirs(GarbageCollectionEnvironment gce,
 
   }
 
-  private boolean getCandidates(GarbageCollectionEnvironment gce, String lastCandidate,
-      List<String> candidates) throws TableNotFoundException {
+  private Iterator<String> getCandidates(GarbageCollectionEnvironment gce)
+      throws TableNotFoundException, IOException {
     try (TraceScope candidatesSpan = Trace.startSpan("getCandidates")) {
-      return gce.getCandidates(lastCandidate, candidates);
+      return gce.getCandidates();
     }
   }
```

Comment:
This method doesn't make sense anymore. Since it's just returning an iterator and not actually doing all the fetching, tracing this operation isn't useful. The trace should be moved to wrap the retrieval of a batch.